### PR TITLE
move to int32 for hit indices

### PR DESCRIPTION
**Pull request by VinInn**

### CODE REVIEW

**Comment by fwyzard** [comment]
:construction: Validation running at fu-c2a02-35-01:/data/user/fwyzard/patatrack/validation/run_583.9KwwVLudje ...

**Comment by fwyzard** [comment]
:construction: Validation running at fu-c2a02-35-01:/data/user/fwyzard/patatrack/validation/run_583.YRNDNOY5Ga ...

**Comment by fwyzard** [comment]
The impact on the throughput for quadruplets is around 1-2%.

No effect on the physics results, as expected.

**Comment by fwyzard** [comment]
I understand the change in data type.
What does `hitToTuple` do ?

**Comment by VinInn** [comment]
```kernel_checkOverflows``` is not run in standard wf
hitToTuple is just a oneToMany association between hits and ntuple; it tells in which tuple a hit is. It is used mainly (only?) by the infamous shred-hit--cleaner (see #552)

in ```kernel_checkOverflows``` is just used to count (and eventually identify) the doublets that do not share any hit with an already built tracks (as requested by you in case lepton reconstruction would require them)

It is experimental code.
This is why I suggested to start from #545 for the hit size issue

File: RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorKernelsImpl.h [hunk 1]
```diff
@@ -95,7 +96,11 @@ __global__ void kernel_checkOverflows(HitContainer const *foundNtuplets,
       atomicAdd(&c.nKilledCells, 1);
     if (0 == thisCell.theUsed)
       atomicAdd(&c.nEmptyCells, 1);
-    if (thisCell.tracks().empty())
+    // if (thisCell.tracks().empty())
+    //  atomicAdd(&c.nZeroTrackCells, 1);
```
Comment (fwyzard) [inline RecoPixelVertexing/PixelTriplets/plugins/CAHitNtupletGeneratorKernelsImpl.h:100]:
```suggestion
```

**Comment by fwyzard** [comment]
Throughput re-checked by hand:
> Running 3 times over 4100 events with 1 jobs, each with 8 threads, 8 streams and 1 GPUs

### master
  - quadruplets: 1136.7 ±   6.4 ev/s
  - triplets: 671.7 ±   2.9 ev/s

### master + #583 
  - quadruplets: 1132.5 ±   4.5 ev/s
  - triplets: 667.6 ±   1.6 ev/s
---
In both cases the drop is less than 1%.

**Pull request merged**